integration/spark2/src/main/scala/org/apache/spark/sql/CarbonDictionaryDecoder.scala

```diff
@@ -175,16 +175,25 @@ case class CarbonDictionaryDecoder(
             val unsafeProjection = UnsafeProjection.create(output.map(_.dataType).toArray)
             var flag = true
             var total = 0L
+
             override final def hasNext: Boolean = iter.hasNext
+
             override final def next(): InternalRow = {
               val startTime = System.currentTimeMillis()
               val row: InternalRow = iter.next()
               val data = row.toSeq(dataTypes).toArray
               dictIndex.foreach { index =>
                 if (data(index) != null) {
-                  data(index) = DataTypeUtil.getDataBasedOnDataType(dicts(index)
-                    .getDictionaryValueForKey(data(index).asInstanceOf[Int]),
-                    getDictionaryColumnIds(index)._3)
+                  try {
+                    data(index) = DataTypeUtil.getDataBasedOnDataType(dicts(index)
+                      .getDictionaryValueForKey(data(index).asInstanceOf[Int]),
+                      getDictionaryColumnIds(index)._3)
+                  }
+                  catch {
+                    case x: ClassCastException => {
```

Comment:
Please just add  `!carbonDimension.isComplex()` in line number 61 and 133. That should solve the problem, remove all other unnecessary excepion handlings